.github/workflows/preview-release.yml

```
@@ -0,0 +1,88 @@
+name: Preview Release
+permissions: {}
+
+on:
+  pull_request:
+    branches: [main]
+    types: [labeled]
+
+# Automatically cancel in-progress actions on the same branch
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event_name == 'pull_request_target' && github.head_ref || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  preview:
+    name: Preview Release
+    if: ${{ github.repository_owner == 'withastro' && github.event.label.name == 'pr-preview' }}
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      pull-requests: write
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@8e8c483db84b4bee98b60c0593521ed34d9990e8 # v6.0.1
+        with:
+          persist-credentials: false
+
+      - name: Setup PNPM
+        uses: pnpm/action-setup@41ff72655975bd51cab0327fa583b6e92b6d3061 # v4.2.0
+
+      - name: Setup Node
+        uses: actions/setup-node@395ad3262231945c25e8478fd5baf05154b1d79f # v6.1.0
+        with:
+          node-version: 24.10.0
+          cache: 'pnpm'
+
+      - name: Install Dependencies
+        run: pnpm i
+
+      - name: Get changeset status
+        id: changeset
+        run: |
+          pnpm changeset status --output changeset-status.json
+          echo "status=$(cat changeset-status.json | jq -c .)" >> $GITHUB_OUTPUT
+
+      - name: Get pnpm packages
+        id: pnpm
+        run: echo "packages=$(pnpm list --recursive --depth -1 --json | jq -c .)" >> $GITHUB_OUTPUT
```

We could filter for public packages only, but I guess since changesets for private packages are not possible, this just makes jq slower:

```suggestion
        run: echo "packages=$(pnpm list --recursive --depth -1 --json | jq -c '[.[] | select(.private == false)]')" >> $GITHUB_OUTPUT
```